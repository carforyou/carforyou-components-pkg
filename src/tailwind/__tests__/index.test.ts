import { defaultConfig, withDefaultConfig } from "../index"

describe("tailwind", () => {
  describe("defaultConfig", () => {
    it("is accessible", () => {
      expect(defaultConfig).toMatchSnapshot()
    })
    it("returns colors", () => {
      expect(defaultConfig.theme.colors.salmon).toEqual("#F73B47")
    })
  })

  describe("withDefaultConfig", () => {
    it("returns withDefaultConfig by default", () => {
      const mergedConfig = withDefaultConfig({})
      expect(mergedConfig).toEqual(defaultConfig)
    })

    it("can override full objects", () => {
      const mergedConfig = withDefaultConfig({})
      mergedConfig.theme.colors = { tuna: "#4E5154" }
      expect(mergedConfig.theme.colors).toEqual({ tuna: "#4E5154" })
    })
    it("can extend objects", () => {
      const colorCount = Object.keys(defaultConfig.theme.colors).length
      const mergedConfig = withDefaultConfig({
        theme: { colors: { tuna: "#4E5154" } },
      })

      expect(Object.keys(mergedConfig.theme.colors)).toHaveLength(
        colorCount + 1
      )
      expect(mergedConfig.theme.colors.tuna).toEqual("#4E5154")
      expect(mergedConfig.theme.colors.salmon).toEqual("#F73B47")
    })

    it("can extend objects for properties that extends colors", () => {
      const borderColorCount = Object.keys(
        defaultConfig.theme.borderColor
      ).length
      const mergedConfig = withDefaultConfig({
        theme: { borderColor: { tuna: "#4E5154" } },
      })

      expect(Object.keys(mergedConfig.theme.borderColor)).toHaveLength(
        borderColorCount + 1
      )

      expect(mergedConfig.theme.borderColor.tuna).toEqual("#4E5154")
      expect(mergedConfig.theme.borderColor.salmon).toEqual("#F73B47")
    })

    it("can extend arrays", () => {
      const pluginCount = Object.keys(defaultConfig.plugins).length
      const mergedConfig = withDefaultConfig({ plugins: [jest.fn()] })

      expect(Object.keys(mergedConfig.plugins)).toHaveLength(pluginCount + 1)
    })
  })
})
