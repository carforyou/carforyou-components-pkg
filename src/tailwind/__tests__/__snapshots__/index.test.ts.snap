// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tailwind defaultConfig is accessible 1`] = `
Object {
  "corePlugins": Object {
    "backgroundOpacity": false,
    "borderOpacity": false,
    "placeholderOpacity": false,
  },
  "plugins": Array [
    [Function],
    [Function],
  ],
  "theme": Object {
    "backgroundColor": Object {
      "black": "#000000",
      "facebook": "#4267B2",
      "fair-deal": "#EAA403",
      "good-deal": "#81B040",
      "great-deal": "#149246",
      "green-bright": "#0DC143",
      "green-dark": "#149246",
      "green-light": "#74CC74",
      "green-opaque": "#E3F4E3",
      "grey-1": "#EBEFF1",
      "grey-2": "#D7DBDF",
      "grey-3": "#A1A7AF",
      "grey-4": "#6F7681",
      "grey-bright": "#F9F9FB",
      "grey-dark": "#232A36",
      "grey-dark-opaque": "#A7AAAF",
      "high-deal": "#E34A15",
      "not-defined": "#A0A7AB",
      "salmon": "#F73B47",
      "salmon-dark": "#ED202D",
      "salmon-light": "#FDE8EA",
      "teal": "#3696B9",
      "teal-bright": "#AFD5E3",
      "teal-dark": "#227D9E",
      "teal-darker": "#1A6072",
      "teal-light": "#D7EAF1",
      "teal-opaque": "#5EABC7",
      "transparent": "transparent",
      "twitter": "#1EA0F2",
      "white": "#FFFFFF",
      "yellow": "#FCB001",
      "yellow-light": "#FEEFCC",
    },
    "backgroundPosition": Object {
      "bottom": "bottom",
      "center": "center",
      "left": "left",
      "left-bottom": "left bottom",
      "left-top": "left top",
      "right": "right",
      "right-bottom": "right bottom",
      "right-top": "right top",
      "top": "top",
    },
    "backgroundSize": Object {
      "contain": "contain",
      "cover": "cover",
    },
    "borderColor": Object {
      "DEFAULT": undefined,
      "black": "#000000",
      "facebook": "#4267B2",
      "fair-deal": "#EAA403",
      "good-deal": "#81B040",
      "great-deal": "#149246",
      "green-bright": "#0DC143",
      "green-dark": "#149246",
      "green-light": "#74CC74",
      "green-opaque": "#E3F4E3",
      "grey-1": "#EBEFF1",
      "grey-2": "#D7DBDF",
      "grey-3": "#A1A7AF",
      "grey-4": "#6F7681",
      "grey-bright": "#F9F9FB",
      "grey-dark": "#232A36",
      "grey-dark-opaque": "#A7AAAF",
      "high-deal": "#E34A15",
      "not-defined": "#A0A7AB",
      "salmon": "#F73B47",
      "salmon-dark": "#ED202D",
      "salmon-light": "#FDE8EA",
      "teal": "#3696B9",
      "teal-bright": "#AFD5E3",
      "teal-dark": "#227D9E",
      "teal-darker": "#1A6072",
      "teal-light": "#D7EAF1",
      "teal-opaque": "#5EABC7",
      "transparent": "transparent",
      "twitter": "#1EA0F2",
      "white": "#FFFFFF",
      "yellow": "#FCB001",
      "yellow-light": "#FEEFCC",
    },
    "borderRadius": Object {
      "10": "10px",
      "2": "2px",
      "20": "20px",
      "4": "4px",
      "6": "6px",
      "8": "8px",
      "full": "9999px",
      "half": "50%",
      "none": "0",
    },
    "borderWidth": Object {
      "0": "0",
      "2": "2px",
      "5": "5px",
      "DEFAULT": "1px",
    },
    "boxShadow": Object {
      "DEFAULT": "0 2px 4px 0 rgba(0,0,0,0.10)",
      "hard": "0 4px 18px 0 rgba(35,42,54,0.4)",
      "none": "none",
      "small": "0 2px 6px 0 rgba(35,42,54,0.1)",
      "soft": "0 4px 18px 0 rgba(35,42,54,0.1)",
    },
    "colors": Object {
      "black": "#000000",
      "facebook": "#4267B2",
      "fair-deal": "#EAA403",
      "good-deal": "#81B040",
      "great-deal": "#149246",
      "green-bright": "#0DC143",
      "green-dark": "#149246",
      "green-light": "#74CC74",
      "green-opaque": "#E3F4E3",
      "grey-1": "#EBEFF1",
      "grey-2": "#D7DBDF",
      "grey-3": "#A1A7AF",
      "grey-4": "#6F7681",
      "grey-bright": "#F9F9FB",
      "grey-dark": "#232A36",
      "grey-dark-opaque": "#A7AAAF",
      "high-deal": "#E34A15",
      "not-defined": "#A0A7AB",
      "salmon": "#F73B47",
      "salmon-dark": "#ED202D",
      "salmon-light": "#FDE8EA",
      "teal": "#3696B9",
      "teal-bright": "#AFD5E3",
      "teal-dark": "#227D9E",
      "teal-darker": "#1A6072",
      "teal-light": "#D7EAF1",
      "teal-opaque": "#5EABC7",
      "transparent": "transparent",
      "twitter": "#1EA0F2",
      "white": "#FFFFFF",
      "yellow": "#FCB001",
      "yellow-light": "#FEEFCC",
    },
    "customRotate": Object {
      "180": "180deg",
      "270": "270deg",
      "90": "90deg",
    },
    "divideColor": [Function],
    "divideWidth": Object {
      "0": "0",
      "2": "2px",
      "5": "5px",
      "DEFAULT": "1px",
    },
    "fontFamily": Object {
      "sans": Array [
        "Muli, Century Gothic, Lucida Grande, Sans-Serif",
      ],
    },
    "fontSize": Object {
      "2xl": "32px",
      "3xl": "48px",
      "base": "16px",
      "lg": "21px",
      "md": "18px",
      "sm": "14px",
      "xl": "24px",
      "xs": "12px",
      "xxs": "10px",
    },
    "fontWeight": Object {
      "bold": 700,
      "regular": 400,
    },
    "height": Object {
      "4": "4px",
      "auto": "auto",
      "checkbox": "24px",
      "full": "100%",
      "input": "52px",
      "rangeFilterChart": "118px",
      "screen": "100vh",
      "scrollbar": "70px",
    },
    "inset": Object {
      "-floatingLabel": "-13px",
      "8": "8px",
      "auto": "auto",
      "floatingLabel": "13px",
      "full": "100%",
      "half": "50%",
      "requiredIndicator": "5px",
    },
    "letterSpacing": Object {
      "normal": "0",
      "wide": "1px",
      "wider": "2px",
    },
    "lineHeight": Object {
      "label": "24px",
      "md": 2,
      "none": 1,
      "sm": 1.5,
      "xs": 1.25,
      "xxs": 0.9,
    },
    "linearGradients": Object {
      "colors": Object {
        "black": Array [
          "rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.5)",
        ],
        "fade": Array [
          "rgba(255,255,255,0) 90%",
          "rgba(255,255,255,1) 100%",
        ],
        "grey": Array [
          "#F9F9FB",
          "#FFFFFF 20%",
        ],
      },
      "directions": Object {
        "to-bottom": "to bottom",
        "to-left": "to left",
        "to-right": "to right",
        "to-top": "to top",
      },
    },
    "margin": Object {
      "-1": "-1px",
      "-10": "-10px",
      "-20": "-20px",
      "-40": "-40px",
      "-45": "-45px",
      "-5": "-5px",
      "-selectMenu": "-21px",
      "-selectWithHintMenu": "-43px",
      "22": "22px",
      "60": "60px",
      "80": "80px",
      "90": "90px",
      "auto": "auto",
      "px": "1px",
    },
    "maxHeight": Object {
      "dropdown": "240px",
      "dropdownSM": "190px",
      "full": "100%",
    },
    "maxWidth": Object {
      "container": "1200px",
      "full": "100%",
    },
    "minHeight": Object {
      "36": "36px",
      "40": "40px",
      "52": "52px",
      "auto": "auto",
      "full": "100%",
      "screen": "100vh",
    },
    "minWidth": Object {},
    "opacity": Object {
      "0": "0",
      "10": ".1",
      "100": "1",
      "20": ".2",
      "40": ".4",
      "60": ".6",
      "80": ".8",
    },
    "padding": Object {
      "13": "13px",
      "14": "14px",
      "3": "3px",
      "8": "8px",
      "px": "1px",
    },
    "screens": Object {
      "lg": "1024px",
      "md": "768px",
      "sm": "576px",
      "xl": "1280px",
      "xs": "480px",
      "xxs": "375px",
    },
    "spacing": Object {
      "0": "0px",
      "10": "10px",
      "12": "12px",
      "15": "15px",
      "16": "16px",
      "20": "20px",
      "25": "25px",
      "3": "3px",
      "30": "30px",
      "40": "40px",
      "5": "5px",
    },
    "textColor": Object {
      "black": "#000000",
      "facebook": "#4267B2",
      "fair-deal": "#EAA403",
      "good-deal": "#81B040",
      "great-deal": "#149246",
      "green-bright": "#0DC143",
      "green-dark": "#149246",
      "green-light": "#74CC74",
      "green-opaque": "#E3F4E3",
      "grey-1": "#EBEFF1",
      "grey-2": "#D7DBDF",
      "grey-3": "#A1A7AF",
      "grey-4": "#6F7681",
      "grey-bright": "#F9F9FB",
      "grey-dark": "#232A36",
      "grey-dark-opaque": "#A7AAAF",
      "high-deal": "#E34A15",
      "not-defined": "#A0A7AB",
      "salmon": "#F73B47",
      "salmon-dark": "#ED202D",
      "salmon-light": "#FDE8EA",
      "teal": "#3696B9",
      "teal-bright": "#AFD5E3",
      "teal-dark": "#227D9E",
      "teal-darker": "#1A6072",
      "teal-light": "#D7EAF1",
      "teal-opaque": "#5EABC7",
      "transparent": "transparent",
      "twitter": "#1EA0F2",
      "white": "#FFFFFF",
      "yellow": "#FCB001",
      "yellow-light": "#FEEFCC",
    },
    "transitionDuration": Object {
      "200": "200ms",
    },
    "width": Object {
      "1/12": "8.33333333%",
      "10/12": "83.33333%",
      "11/12": "91.66666667%",
      "12/12": "100%",
      "2/12": "16.66667%",
      "3/12": "25%",
      "4/12": "33.33333%",
      "5/12": "41.66666667%",
      "6/12": "50%",
      "7/12": "58.33333333%",
      "8/12": "66.66667%",
      "9/12": "75%",
      "auto": "auto",
      "checkbox": "24px",
      "clearButton": "32px",
      "modal": "570px",
      "modalLarge": "770px",
      "modalSmall": "370px",
      "screen": "100vw",
      "scrollbar": "14px",
    },
    "zIndex": Object {
      "1": 1,
      "card": 10,
      "dropdownMenu": 20,
      "flashMessage": 99,
      "intercom": 10,
      "modal": 98,
      "modalClose": 10,
      "negative": -1,
    },
  },
  "variants": Object {
    "alignContent": Array [
      "responsive",
    ],
    "alignItems": Array [
      "responsive",
    ],
    "alignSelf": Array [
      "responsive",
    ],
    "appearance": Array [
      "responsive",
    ],
    "backgroundAttachment": Array [
      "responsive",
    ],
    "backgroundColor": Array [
      "responsive",
      "hover",
      "focus",
      "active",
    ],
    "backgroundPosition": Array [
      "responsive",
    ],
    "backgroundRepeat": Array [
      "responsive",
    ],
    "backgroundSize": Array [
      "responsive",
    ],
    "borderCollapse": Array [
      "responsive",
    ],
    "borderColor": Array [
      "responsive",
      "hover",
      "focus",
      "focus-within",
    ],
    "borderRadius": Array [
      "responsive",
    ],
    "borderStyle": Array [
      "responsive",
    ],
    "borderWidth": Array [
      "responsive",
      "hover",
    ],
    "boxShadow": Array [
      "responsive",
      "hover",
      "focus",
    ],
    "cursor": Array [
      "responsive",
    ],
    "display": Array [
      "responsive",
    ],
    "divideColor": Array [
      "responsive",
    ],
    "divideWidth": Array [
      "responsive",
    ],
    "fill": Array [
      "responsive",
    ],
    "flex": Array [
      "responsive",
    ],
    "flexDirection": Array [
      "responsive",
    ],
    "flexGrow": Array [
      "responsive",
    ],
    "flexShrink": Array [
      "responsive",
    ],
    "flexWrap": Array [
      "responsive",
    ],
    "float": Array [
      "responsive",
    ],
    "fontFamily": Array [
      "responsive",
    ],
    "fontSize": Array [
      "responsive",
    ],
    "fontSmoothing": Array [
      "responsive",
    ],
    "fontStyle": Array [
      "responsive",
    ],
    "fontWeight": Array [
      "responsive",
      "hover",
      "focus",
    ],
    "gap": Array [
      "responsive",
    ],
    "height": Array [
      "responsive",
    ],
    "inset": Array [
      "responsive",
    ],
    "justifyContent": Array [
      "responsive",
    ],
    "letterSpacing": Array [
      "responsive",
    ],
    "lineHeight": Array [
      "responsive",
    ],
    "listStylePosition": Array [
      "responsive",
    ],
    "listStyleType": Array [
      "responsive",
    ],
    "margin": Array [
      "responsive",
    ],
    "maxHeight": Array [
      "responsive",
    ],
    "maxWidth": Array [
      "responsive",
    ],
    "minHeight": Array [
      "responsive",
    ],
    "minWidth": Array [
      "responsive",
    ],
    "objectFit": Array [
      "responsive",
    ],
    "objectPosition": Array [
      "responsive",
    ],
    "opacity": Array [
      "responsive",
      "hover",
    ],
    "order": Array [
      "responsive",
    ],
    "outline": Array [
      "responsive",
      "focus",
    ],
    "overflow": Array [
      "responsive",
    ],
    "padding": Array [
      "responsive",
    ],
    "pointerEvents": Array [
      "responsive",
    ],
    "position": Array [
      "responsive",
    ],
    "resize": Array [
      "responsive",
    ],
    "scale": Array [
      "responsive",
      "hover",
      "focus",
    ],
    "skew": Array [
      "responsive",
      "hover",
      "focus",
    ],
    "space": Array [
      "responsive",
    ],
    "stroke": Array [
      "responsive",
    ],
    "tableLayout": Array [
      "responsive",
    ],
    "textAlign": Array [
      "responsive",
    ],
    "textColor": Array [
      "responsive",
      "hover",
      "focus",
      "focus-within",
    ],
    "textDecoration": Array [
      "responsive",
      "hover",
      "focus",
    ],
    "textOverflow": Array [
      "responsive",
    ],
    "textTransform": Array [
      "responsive",
    ],
    "transitionDuration": Array [
      "responsive",
      "hover",
    ],
    "transitionProperty": Array [
      "responsive",
      "hover",
    ],
    "transitionTimingFunction": Array [
      "responsive",
      "hover",
    ],
    "translate": Array [
      "responsive",
      "hover",
      "focus",
    ],
    "userSelect": Array [
      "responsive",
    ],
    "verticalAlign": Array [
      "responsive",
    ],
    "visibility": Array [
      "responsive",
    ],
    "whitespace": Array [
      "responsive",
    ],
    "width": Array [
      "responsive",
    ],
    "wordBreak": Array [
      "responsive",
    ],
    "zIndex": Array [
      "responsive",
      "hover",
      "focus",
    ],
  },
}
`;
